SimpleMap/LocationDataManager.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  ViewController.swift
+//  SimpleMap
+//
+//  Created by Artem Eliseev on 05/07/2023.
+//
+
+import Foundation
+import CoreLocation
+import UIKit
+import MapKit
+
+
+class LocationDataManager : NSObject, ObservableObject, CLLocationManagerDelegate {
+    var locationManager = CLLocationManager()
+    @Published var authorizationStatus: CLAuthorizationStatus?
+    @Published var location: CLLocation?
+    @Published var region = MKCoordinateRegion(
+        center: CLLocationCoordinate2D(latitude: 51, longitude: 0),
+        span: MKCoordinateSpan(latitudeDelta: 10, longitudeDelta: 10)
+    )
+    @Published var hasSetRegion = false
```

Comment:
it's best practice to think about the 'open/closed' principal by encapsulating the logic of a class privately so it's not accessible outside.. in this case think about which of these properties should be marked as `private var ...` 

also should they all be `var` or should some be `let` which is constant/non-changing values?